src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -456,6 +466,15 @@ seastar::future<> HTTPProxy::gracefulStop() {
 seastar::future<> HTTPProxy::start() {
     _registerMetrics();
     _registerAPI();
+    _expiryList.start(httpproxy_txn_timeout(), [this](ManagedTxn& txn) {
+        auto ts = txn.timestamp;
+        K2LOG_I(log::httpproxy, "Removing txn {} because of timeout", ts);
+        return txn.handle.kill()
+            .then([this, ts]{
```

Comment:
I think there maybe a range of race conditions with the list and _txns here. I think we'd have to move the handle out of _txns (i.e. extract() it from the map) and then call kill on it